src/RamCondition.h

```diff
@@ -282,8 +282,8 @@ class RamAbstractExistenceCheck : public RamCondition {
      *  A null pointer element in the vector denotes an unspecified
      *  pattern for a tuple element.
      */
-    const std::vector<RamExpression*> getValues() const {
-        return toPtrVector(values);
+    const std::vector<std::unique_ptr<RamExpression>>& getValues() const {
+        return values;
```

Comment:
If you could fix the bitcasts without changing the RAM interface it would be much better.

The interface in general should be either the underlying smart pointer vectors or raw pointers that hide that hide the underlying structure, not a mix of both. Changing this interface should involve some discussions first.

Reply:
I've rebased and cleaned that out.